k2/python/k2/ragged/tensor.py

```diff
@@ -42,14 +42,18 @@ def __init__(self,
         '''
         if isinstance(ragged, str):
             ragged = _k2.RaggedFloat(ragged)
+            assert values is None
         elif isinstance(ragged, _k2.RaggedShape):
             assert values is not None
             ragged = _k2.RaggedFloat(ragged, values)
 
         assert isinstance(ragged, _k2.RaggedFloat)
 
         self.ragged = ragged
-        self._scores = ragged.values()
+        if values is not None:
+            self._scores = values
```

Comment:
Wait, why does RaggedFloat have a _scores member?

Reply:
... it is supposed to be a generic interface, not tied into class Fsa.

Reply:
> Wait, why does RaggedFloat have a _scores member?

While doing `normalize_scores`, if we use `_k2.RaggedFloat` directly it is not possible to register the
operation into the autograd graph since there is no `torch.Tensor`. 

`k2.ragged.RaggedFloat` is a wrap around `_k2.RaggedFloat` and the field `_scores` is to make
autograd work.

https://github.com/k2-fsa/k2/blob/75f9378c3bae8c01a280a32cb4c2f6afb938951a/k2/python/k2/ragged/autograd.py#L15-L16

For the autograd to work, we need to pass a `torch.Tensor` to it. One alternative is to let the user pass a tensor
while calling `normalize_scores`.

Reply:
> ... it is supposed to be a generic interface, not tied into class Fsa.

The class `RaggedFloat` is not tied to `class Fsa`; it is a wrapper around `_k2.RaggedFloat`.

Reply:
I think the name _values would be better than _scores; we don't want to imply the values have any specific meaning.

Reply:
will change it.

Reply:
And please make sure the documentation of class RaggedFloat is as self-contained as possible and explains anything that needs to be explained about autograd...  (it may already be; just a reminder).

Reply:
I am adding more documentation for it.